### ISSUE TITLE
content field in parameter part of an openapi specification not supported

### ISSUE THREAD

**Issue by axelsautter** [zaproxy/zaproxy#6166]
zap-api-scan.py cannot handle Query parameters with content tag ( Parameters with content are supported in Swagger UI 3.23.7+ and Swagger Editor 3.6.34+. ).
As result, APIs generated with these swagger versions cannot be scanned with zap-api-scan.py.
References:
- [Swagger serialization](https://swagger.io/docs/specification/serialization/)
- [Swagger schema versus content](https://swagger.io/docs/specification/describing-parameters/#schema-vs-content)

Support of the content tag is needed to ensure further usage of zap for swagger generated APIs.

Currently scanning those api-definitions results in:
```
10329 [ZAP-Import-OpenAPI-1] WARN org.zaproxy.zap.extension.openapi.ExtensionOpenApi  -
java.lang.NullPointerException
    at org.zaproxy.zap.extension.openapi.generators.DataGenerator.generateDefaultValue(DataGenerator.java:79)
    at org.zaproxy.zap.extension.openapi.generators.DataGenerator.generate(DataGenerator.java:69)
    at org.zaproxy.zap.extension.openapi.generators.PathGenerator.generateFullPath(PathGenerator.java:42)
    at org.zaproxy.zap.extension.openapi.converter.swagger.RequestModelConverter.generatePath(RequestModelConverter.java:55)
    at org.zaproxy.zap.extension.openapi.converter.swagger.RequestModelConverter.convert(RequestModelConverter.java:41)
    at org.zaproxy.zap.extension.openapi.converter.swagger.SwaggerConverter.convertToRequest(SwaggerConverter.java:164)
    at org.zaproxy.zap.extension.openapi.converter.swagger.SwaggerConverter.getRequestModels(SwaggerConverter.java:158)
    at org.zaproxy.zap.extension.openapi.ExtensionOpenApi$3.run(ExtensionOpenApi.java:235)
```

Example parameter:
Parameter example (SNSSAI):
```
paths:
  /pcfBindings:
    post:
      requestBody:
           .
           .
      responses:
        '201':
          .
          .
    get:
      parameters:
        - name: snssai
          in: query
          description: The identification of slice.
          required: false
          content:
            application/json:
          schema:
            $ref: '/zap/wrk/TS29571_CommonData.yaml#/components/schemas/Snssai'
        - name: ipDomain
          in: query
          description: The IPv4 address domain identifier.
          required: false
          schema:
            type: string
          .
          . 
```

Thanks and best regards
/Axel

**Comment by MathieuDV-hpe** [comment]
Same trouble here, with a slightly different error (obtained with the weekly Docker image, by calling zap-api-scan.py):
![image](https://user-images.githubusercontent.com/80677795/111130533-9d348780-8577-11eb-8192-ea53c40cfbdb.png)

We found the same source for this bug (content field in parameter section), on json openapi definitions. Could we have some updates or possible workarounds ?

**Comment by thc202** [comment]
@MathieuDV-hpe could you share your definition (redacted as needed)? Are you also using refs for the schema?

**Comment by WhyJee** [comment]
@thc202 we made some deeper analysis and narrowed down a bit the problem analyzing URI one by one.


I narrowed the analysis just to the GetIdentityData URI "[/subscription-data/{ueId}/identity-data](https://github.com/jdegre/5GC_APIs/blob/master/TS29505_Subscription_Data.yaml#L4254)" which has only two parametrers:
* UE ID
* Application Port Identifier

I thought it was related to the default missing, or looking at the wrong code, but looking more closely at the faulty code
https://github.com/zaproxy/zap-extensions/blob/8d18127bdf10f1cade42d8b3aa2758b8af38ab2f/addOns/openapi/src/main/java/org/zaproxy/zap/extension/openapi/generators/DataGenerator.java#L68

makes me think (line 74) that it was indeed the schema that was null and this explained the NPE in line #85.

So looking back at the definition of the parameter started to bring some lights:
```json
{
   "content": {
     "application/json": {
       "schema": {
         "$ref": "#/components/schemas/AppPortId"
       }
     }
   },
   "description": "Application port identifier",
   "in": "query",
   "name": "app-port-id"
},
```
In fact this seemed not a parameter definition, and will generally be found in response definition.
I had a bad feeling with the "application/json" line here and thus tried to eliminate it with no result of course.

Thus I completely removed the parameter and the test ran successfully (at least with no error).

So I checked the openapi syntax ending to https://swagger.io/specification/#operation-object which contains not "content" field. And this explain the missing schema at that point. To confirm this I also removed the "content" level having directly the "schema" as is supported and this is ok also confirming the theory... almost --> Wrong analysis 😁 

Content is indeed supported by Openapi spec as is explained in [Describing Parameters](https://swagger.io/docs/specification/describing-parameters/):

> To describe a parameter, you specify its `name`, location (`in`), data type (defined by either `schema` or **`content`**) and other attributes, such as `description` or `required`.

Details from [schema vs content](https://swagger.io/docs/specification/describing-parameters/#schema-vs-content) indicates the support:
> Note for Swagger UI and Swagger Editor users: Parameters with content are supported in Swagger UI 3.23.7+ and Swagger Editor 3.6.34+.

In summary, what is missing is the support of the `content` data type.
